### ISSUE TITLE
IllegalStateException: Feature order cycle found, involved sources: ars_elemental:flashing_forest

### ISSUE THREAD

**Issue by frquadrat** [FTBTeam/FTB-Modpack-Issues#3528]
### Modpack

FTB Presents Direwolf20 1.19

### Modpack version

1.12.0

### Log Files

https://gist.github.com/frquadrat/9bc775c4e07f1623df8a97db4829e335

### Describe the bug

forge server running on a linux system.

I created a RFToolsDimension and am using an RFToolsBuilder as a quarry/pump.

At one point the server stopped running and any restart fails with the same issue:

``` [Server thread/ERROR] [net.minecraft.util.thread.BlockableEventLoop/FATAL]: Error executing task on Chunk source main thread executor for rftoolsdim:frr_hf2
net.minecraft.ReportedException: Exception generating new chunk
[..]
Caused by: java.lang.IllegalStateException: Feature order cycle found, involved sources: [Reference{ResourceKey[minecraft:worldgen/biome / ars_elemental:flashing_forest]=net.minecraft.world.level.biome.Biome@691d150b}]
```
Not sure if that helps: I added [cyanide-forge](https://github.com/alcatrazEscapee/cyanide) and that reports the following:
```
Caused by: com.alcatrazescapee.cyanide.codec.FeatureCycleDetector$FeatureCycleException: A feature cycle was found.

Cycle:
At step 9
Feature 'byg:global/vegetal_decoration'
  must be before 'quark:deferred_feature_vegetal_decoration' (defined in 'byg:red_rock_valley' at index 10, 11 and 183 others)
  must be before 'waystones:waystone' (defined in 'byg:red_rock_valley' at index 11, 13 and 174 others)
  must be before 'farmersdelight:patch_wild_onions' (defined in 'quark:glimmering_weald' at index 11, 13 and 46 others)
  must be before 'supplementaries:wild_flax' (defined in 'quark:glimmering_weald' at index 13, 15 and 48 others)
  must be before 'ars_nouveau:archwood' (defined in 'byg:red_rock_valley' at index 14, 15 and 112 others)
  must be before 'integrateddynamics:tree_menril_general' (defined in 'byg:red_rock_valley' at index 15, 16 and 112 others)
  must be before 'ars_elemental:common_flashing_archwood' (defined in 'ars_nouveau:archwood_forest' at index 19, 20 and 1 others)
  must be before 'ars_nouveau:patch_grass_forest' (defined in 'ars_elemental:flashing_forest' at index 8, 9)
  must be before 'ars_nouveau:flower_default' (defined in 'ars_elemental:flourishing_forest' at index 9, 10 and 4 others)
  must be before 'byg:global/vegetal_decoration' (defined in 'ars_elemental:flourishing_forest' at index 10, 11 and 4 others)
```

What can I do in order to make it work again? (For now I disabled the loading of the chunk the RFToolsBuilder resides in so that the separate dimension isn't being loaded.)
Where does the issue reside in? Is it configuration or is it Ars Elemental?

Thanks a lot
FR²

### Steps to reproduce

Difficult to reproduce. I could provide the world or parts of it, if that's helpful.
Maybe: 
-Create a stable RFToolsDimension with only one biome (DeepOcean), only one block, liquid mekanism:liquid_fluoric_acid, ..
- place an rftoolsbuilder with a pump card with the largest possible dimensions (512:256:512) and let it go to work, i.e. it will load/generate the chunks it needs to access.
- at one point or another the described issue will kill the server

### Expected behaviour

the server should run without issues

**Comment by seschi98** [comment]
@frquadrat did you figure out a solution for this issue? I also encountered the same problem and my server crashes once in a while

**Comment by Zakru** [comment]
Same error rendered our server unusable. I made my second fully randomized RFTDim dimension (only digit 1 dimlet), and as the dimension builder froze at 99% the server crashed. After restarting the dim tab said something like -1 power so I just emptied it and made a digit 2 dim. Now after some time, the server crashed permanently on startup, stating that the error above occurred while generating a chunk in that same dimension.

Dumping with `nbted` shows this as the random dimlets

    String "randomized" "[{\"t\":\"b\",\"n\":\"thermal:machine_frame\"},{\"t\":\"t\",\"n\":\"ravine\"},{\"t\":\"s\",\"n\":\"default\"},{\"t\":\"bt\",\"n\":\"mushroom\"},{\"t\":\"bc\",\"n\":\"default\"},{\"t\":\"ti\",\"n\":\"normal\"}]"

I tried deleting the dimension data, which seems to allow the server to launch and stay on at least for a while, but I kept getting that error in the console when the game tries to save the dimension, even after `/rftoolsdim forget`.

After deleting the dimension from `level.dat` -> `Data` -> `WorldGenSettings` or something, the error is no longer appearing.
